application/electron/src/classes/class.statefile.ts

```diff
@@ -97,50 +106,56 @@ export class StateFile<TState> implements IService {
             FS.readTextFile(this._file).then((content: string) => {
                 const state = Objects.getJSON(content);
                 if (state instanceof Error) {
-                    this._logger.error(`Cannot parse state file "${this._file}" due error: ${state.message}. Content: "${content}"`);
                     if (!this._allowResetToDefault) {
                         return reject(new Error(`Fail to get JSON from content of "${this._file}" due error: ${state.message}`));
                     } else {
+                        this._logger.warn(`Cannot parse state file "${this._file}" due error: ${state.message}. Content: "${content}". File would be dropped to defaults values.`);
                         return this._default(true).then(() => {
                             resolve(this._defaults);
                         }).catch(reject);
                     }
                 }
-                const valid: Error | void = this._validate(state);
-                if (valid instanceof Error && !this._allowResetToDefault) {
-                    return reject(new Error(this._logger.error(`Wrong format of state file "${this._file}": ${valid.message}`)));
-                } else if (valid instanceof Error) {
-                    this._logger.error(`Wrong format of state file "${this._file}": ${valid.message}.\n File will be reset to defualts values.`);
-                    return this._default(true).then(() => {
-                        resolve(this._defaults);
-                    }).catch(reject);
-                }
-                resolve(state);
+                resolve(this._validate(state));
             }).catch((error: Error) => {
-                this._logger.error(`Fail to read state at "${this._file}" due error: ${error.message}`);
+                reject(this._logger.error(`Fail to read state at "${this._file}" due error: ${error.message}`));
```

Comment:
Ah this is also my usual error ) Weird why TS doesn't check it :/
`reject` always should return `Error` object. Because on catch we are always expecting `Error`, but not a `string`
Well, it would be better
```
reject(new Error(this._logger.error(`Fail to read state at "${this._file}" due error: ${error.message}`)));
```